### ISSUE TITLE
Ask for the details to reproduce the two-stage training

### ISSUE THREAD

**Issue by 5RJ** [ivonajdenkoska/tulip#7]
Hi, thanks for your great work. I'm interested on the training process. I've tried to reimplement the two stage training, however, the precision is much lower than the released one, so could you please release more training details?
Here is my training details.
**stage1: distill**
_train_data_: in total: 2491802 with fixed [sharegpt_prepocessing.py] (https://github.com/ivonajdenkoska/tulip/issues/6#issuecomment-2656089264)
_val_data_: intotal: 2000
_use gpu_: 4 H100.
_the cmd is as follows:_ I change the _student-context-length_ into 77, otherwise it will lead to error; I change the batch-size and accum-freq, but make the total bsz=640.
gpu_num=0,1,2,3
torchrun -m \
    --nnodes=1 \
    --nproc_per_node=${gpu_num} \
    --master_port=2501 \
    training.main_distill_rope \
    --dataset-type "csv" \
    --train-data "/home/jovyan/LMM/wrj/dataset/ShareGPT4V/data/tulip_train.csv" \
    --val-data "/home/jovyan/LMM/wrj/dataset/ShareGPT4V/data/tulip_val.csv" \
    --log_dir "./logs/sharegpt4v/" \
    --warmup 1000 \
    --lr 5e-4 \
    --wd 0.1 \
    --epochs 25 \
    --workers 8 \
    --batch-size 160 \
    --save-frequency 5 \
    --model "ViT-L-14" \
    --pretrained "openai" \
    --precision 'amp_bf16' \
    --log-every-n-steps 100 \
    --accum-freq 1 \
    --context-length 77 \
    --student-context-length 77 \
    --wandb-project-name "dense-cap-distill" \
    --loss-type "cosine" \
    --report-to "wandb"

The loss at epoch 20 is: (the 'epoch' in img is start from 0.)
![Image](https://github.com/user-attachments/assets/f8a7aa72-9a83-456a-b941-2979bd49c856)

**stage2: finetune**

_train_data_: the same as 1st stage
_val_data_: the same as 1st stage
_use gpu_: 4 H100.
_the cmd is as follows_: I change the batch-size and accum-freq, but make the total bsz=1280; I use the _epoch20_ as the pretrained model.
gpu_num=0,1,2,3
torchrun -m \
    --nnodes=1 \
    --nproc_per_node=${gpu_num} \
    --master_port=${master_port} \
    training.main_context_finetune_rope \
    --dataset-type "csv" \
    --batch-size 40 \
    --train-data "/home/jovyan/LMM/wrj/dataset/ShareGPT4V/data/tulip_train.csv" \
    --val-data "/home/jovyan/LMM/wrj/dataset/ShareGPT4V/data/tulip_val.csv" \
    --log_dir "./logs/sharegpt4v/" \
    --warmup 1000 \
    --lr 1e-5 \
    --wd 0.1 \
    --epochs 3 \
    --workers 8 \
    --save-frequency 1 \
    --model "ViT-L-14" \
    --pretrained "openai" \
    --precision 'amp_bf16' \
    --log-every-n-steps 100 \
    --accum-freq 8 \
    --context-length 77 \
    --student-context-length 248 \
    --loss-type "clip_loss" \
    --student-model "/home/jovyan/LMM/wrj/workspace/project/LongClip/tulip/open_clip/logs/sharegpt4v/2025_02_11-09_59_11-distill_RoPE_ViT-L-14-lr_0.0005-b_160-j_8-p_amp_bf16-CTX_77-loss_cosine/checkpoints/epoch_20.pt" \
    --wandb-project-name "dense-cap-ctx-extension" \
    --report-to "wandb"

And the loss of epoch 1 is as follows:

![Image](https://github.com/user-attachments/assets/903ddb7d-daae-4c11-af4a-eee603841650)

Then, I test the final model, the precision is as follows: (BTW, I also couldn't reproduce the eval precision of the released model. ref: https://github.com/ivonajdenkoska/tulip/issues/4#issuecomment-2653824873)

![Image](https://github.com/user-attachments/assets/0d64fcf7-3ca6-428a-b445-3a03203190d0)
the test cmd is:
export CUDA_VISIBLE_DEVICES=1 && python eval_tulip.py --model_name ViT-L-14 --pretrained epoch_1.pt --distilled_model_path epoch_1.pt --pos_encodings rope --context_length 248 --run_urban1k --run_flickr --run_coco

**Comment by 5RJ** [comment]
Maybe the environment difference will lead to such gap? could you please release the specific version of packages in your environment?
